File: python-improperoutputhandling/secure7.py
from openai import OpenAI

client = OpenAI()


def generate_report(data_summary):
    response = client.chat.completions.create(
        model="gpt-4o-mini",
        messages=[{"role": "user", "content": f"Write a plain text report for: {data_summary}"}],
    )
    report = response.choices[0].message.content
    with open("report.txt", "w") as f:
        f.write(report)
    return report


if __name__ == "__main__":
    import sys
    print(generate_report(sys.argv[1]))

File: typescript-misinformation/secure7.ts
import express, { Request, Response } from "express";
import OpenAI from "openai";

const app = express();
app.use(express.json());
const openai = new OpenAI();
const summaries: string[] = [];

app.post("/summarize", async (req: Request, res: Response) => {
  const { text } = req.body as { text: string };
  const completion = await openai.chat.completions.create({
    model: "gpt-4o",
    messages: [{ role: "user", content: "Summarize: " + text }],
  });
  summaries.push(completion.choices[0].message.content!);
  res.json({ status: "stored", count: summaries.length });
});

app.listen(3000);
